### PR TITLE
Replace byteorder with native conversion

### PR DESCRIPTION
**Pull request by HDauven**
## Summary

- replace the only `byteorder` call with `u64::from_le_bytes`
- remove the `byteorder` dependency and Cargo feature
- make `pippenger` available whenever `groups` and `alloc` are enabled
- add committed compatibility tests for radix boundaries, legacy compressed outputs, and scalar edge values

This is a draft because removing the public Cargo feature can break downstream manifests that explicitly request `byteorder`, even though no Dusk repository does so.

## Validation

- `RUSTUP_TOOLCHAIN=1.96.1 make test`
- verify Pippenger against a naïve sum at the 499/500 and 799/800 boundaries
- verify hard-coded compressed outputs generated by the old implementation at those boundaries
- verify native little-endian limbs for zero, one, and the maximum canonical scalar
- `byteorder` is absent from Cargo metadata and the dependency tree

## Performance and size

Pippenger benchmark results ranged from -3.34% to +2.50%, with no consistent trend. Median clean release compile time changed from 5.108 s to 5.093 s (-0.28%). Final binary code/data size was identical at 411,581 bytes; the library artifact was 1,054 bytes smaller.

### CODE REVIEW

File: src/dusk/multiscalar_mul.rs [hunk 1]
```diff
@@ -9,7 +9,6 @@ use alloc::vec::*;
 /// Performs multiscalar multiplication reliying on Pippenger's algorithm.
```
[the remark this answers is not in the document]
Comment (HDauven) [reply src/dusk/multiscalar_mul.rs]: Fixed in 44fd2a2.

**Review by Neotamandua — APPROVED** [review]
LGTM
